File: solutions/csharp/loggingframework/LogMessage.cs
using System;

namespace LoggingFramework
{
    public class LogMessage
    {
        public LogLevel Level { get; }
        public string Message { get; }
        public long Timestamp { get; }

        public LogMessage(LogLevel level, string message)
        {
            Level = level;
            Message = message;
            Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        }

        public override string ToString()
        {
            return $"[{Level}] {Timestamp} - {Message}";
        }
    }
}

File: solutions/java/src/loggingframework/LogMessage.java
package loggingframework;

public class LogMessage {
    private final LogLevel level;
    private final String message;
    private final long timestamp;
    private final String threadName;

    public LogMessage(LogLevel level, String message) {
        this.level = level;
        this.message = message;
        this.timestamp = System.currentTimeMillis();
        this.threadName = Thread.currentThread().getName();
    }

    public LogLevel getLevel() {
        return level;
    }

    public String getMessage() {
        return message;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public String getThreadName() {
        return threadName;
    }

    @Override
    public String toString() {
        return "[" + level + "] " + timestamp + " - " + message;
    }
}
